go/vt/schema/ddl_strategy.go

```diff
@@ -99,20 +76,56 @@ func ParseDDLStrategy(strategyVariable string) (*DDLStrategySetting, error) {
 		setting.Options = submatch[2]
 	}
 
-	switch strategy := DDLStrategy(strategyName); strategy {
-	case "": // backward compatiblity and to handle unspecified values
-		setting.Strategy = DDLStrategyDirect
-	case DDLStrategyVitess, DDLStrategyOnline, DDLStrategyGhost, DDLStrategyPTOSC, DDLStrategyMySQL, DDLStrategyDirect:
-		setting.Strategy = strategy
-	default:
-		return nil, fmt.Errorf("Unknown online DDL strategy: '%v'", strategy)
+	var err error
+	setting.Strategy, err = ParseDDLStrategyName(strategyName)
+	if err != nil {
+		return nil, err
 	}
 	if _, err := setting.CutOverThreshold(); err != nil {
 		return nil, err
 	}
 	return setting, nil
 }
 
+func ParseDDLStrategyName(name string) (tabletmanagerdatapb.OnlineDDL_Strategy, error) {
+	if name == "" {
+		// backward compatiblity and to handle unspecified values
+		return tabletmanagerdatapb.OnlineDDL_DIRECT, nil
+	}
+
+	lowerName := strings.ToUpper(name)
```

Comment:
`lowerName := strings.ToUpper(...)` is contradictory is confusing. Let's do use `ToLower()` and look for texts like `"gh-ost"` instread of `"GH-OST"`

```suggestion
	lowerName := strings.ToLower(name)
```

Reply:
ah it should actually be `upperName` because we need the ALL_CAPS to index into the enum table